### DISCUSSION TITLE
Getting started on Windows 10 (latest experience)

### DISCUSSION THREAD

**Discussion by etherbrat** [micropython/micropython#10070] [Using MicroPython]
Old documentation for Windows 10 seems to rely on DfuSeDemo (an early STM DFU programmer) and an unsigned USB serial driver. The latter is particularly problematic if Bitlocker is enabled on the PC.

Try the following instead.

Download and install STM32CubeProgrammer. Note that STM32CubeProgrammer can handle a range of file formats but NOT .dfu.

Installing STM32CubeProgrammer also seems to install suitable drivers for:
* DFU programming
* the PyBoard MicroPython file system (via USB), and
* the PyBoard MicroPython REPL (via USB serial)

You may also need to download and install a package which contains the aforementioned DfuSeDemo (NOT USED) and Dfu File Manager (which can convert between various file formats).

Download the latest MicroPython firmware, which is published in .dfu and .hex formats. If you like, convert the .dfu file to .s19 (tested) or just use the .hex file (not tested).

Install the DFU programming jumper on the PyBoard, plug in the USB cable (for both power and USB communications) and burn the firmware using STM32CubeProgrammer (Recommended: Verify the download; and do NOT "run" the software).

Unplug the USB cable, remove the DFU programming jumper from the PyBoard. Replug the USB cable, and everything should work.

**Comment by jimmo** [comment]
Thanks @etherbrat 

> or just use the .hex file (not tested).

This does work (making it possible to use CubeProgrammer was a big part of the reason why we started publishing .hex files for stm32 builds).

(It's still very confusing to me that CubeProgrammer doesn't support .dfu files though!!)

> Install the DFU programming jumper on the PyBoard

You can also use `machine.bootloader()` at the REPL.

> Old documentation for Windows 10

We should fix this -- which documentation were you looking at?

> Installing STM32CubeProgrammer also seems to install suitable drivers for:
> DFU programming
> the PyBoard MicroPython file system (via USB), and
> the PyBoard MicroPython REPL (via USB serial)

My understanding is that the last two (file system and USB serial) do not require _any_ drivers to be installed (the filesystem has never required drivers, and the serial not since Windows 10 at least)? Is that not the case?

**Reply by etherbrat** [reply]
Hi. Your queries in order:
* I figured that you had added .hex format files for this reason.
* machine.bootloader() sounds like an easy thang - I had to solder header pins onto my board, just so that I could install a jumper, albeit there will be more to solder later, when I decide exactly how to connect the external electronics (so many choices!).
* Old documentation: I had one document called something like "The care are and feeding of Pythons in the Redmond Zoo", which admittedly was never for Windows 10, and another (attached, referencing the first) that I cobbled together myself from various sources, including (probably) forum posts - this latter worked ok on another computer that didn't have Bitlocker enabled.
* USB drivers: Non-expert here. All I can say is that things started working automagically after I installed STM32CubeProgrammer - except that in my case DfuseDemo -stopped- working after installed the STM32CubeProgrammer, which was unexpected.
* Incidentally, all of this was something of a diving save, after I had all-but decided to abandon Windows 10 and do my PyBoard development on Ubuntu in a VirtualBox VM - all of my accumulated Linux wisdom still worked perfectly. I noticed something about STM32CubeProgrammer out of the corner of my eye and thought I would give it a try. The only Linux glitch was that Putty was breaking due to a missing font setup - easily fixed in the end. Not that I prefer windows, but it is just that little bit faster than Ubuntu in a VM.
[SettingUpThePyboardUnderWindows10.pdf](https://github.com/micropython/micropython/files/10089708/SettingUpThePyboardUnderWindows10.pdf)
